packages/firebase_admob/ios/Classes/FLTMobileAd.m

```diff
@@ -101,70 +101,108 @@ - (NSString *)description {
-
-      [self.adView.leftAnchor constraintGreaterThanOrEqualToAnchor:guide.leftAnchor],
-      [self.adView.rightAnchor constraintLessThanOrEqualToAnchor:guide.rightAnchor],
-    ]];
-
-    if (_anchorType == 0) {
-      [NSLayoutConstraint activateConstraints:@[
-        [self.adView.bottomAnchor constraintEqualToAnchor:guide.bottomAnchor
-                                                 constant:_anchorOffset],
-      ]];
-    } else {
-      [NSLayoutConstraint activateConstraints:@[
-        [self.adView.topAnchor constraintEqualToAnchor:guide.topAnchor constant:_anchorOffset],
-      ]];
-    }
+    [self activateConstraintForLayoutGuide:parentView.safeAreaLayoutGuide];
+    return;
+  } else if (@available(ios 9.0, *)) {
+    [self activateConstraintForLayoutGuide:parentView.layoutMarginsGuide];
+    return;
   }
-#endif
 
-  // We find the left most point that aligns the view to the horizontal center.
-  CGFloat x =
-      screen.frame.size.width / 2 - self.adView.frame.size.width / 2 + _horizontalCenterOffset;
-  // We find the top point that anchors the view to the top/bottom depending on anchorType.
-  CGFloat y;
+  NSLayoutAttribute verticalAttribute = 0;
+  CGFloat anchorConstant = 0.0;
   if (_anchorType == 0) {
-    y = screen.frame.size.height - self.adView.frame.size.height + _anchorOffset;
+    verticalAttribute = NSLayoutAttributeBottom;
+    anchorConstant = -_anchorOffset;
+  } else if (_anchorType == 1) {
+    verticalAttribute = NSLayoutAttributeTop;
+    anchorConstant = _anchorOffset;
+  }
+
+  [self activateConstraintParentView:parentView verticalAttribute:verticalAttribute];
+}
+
+- (void)activateConstraintForLayoutGuide:(UILayoutGuide *_Nonnull)layoutGuide
+    API_AVAILABLE(ios(9.0)) {
+  self.view.translatesAutoresizingMaskIntoConstraints = NO;
+
+  NSLayoutConstraint *verticalConstraint = nil;
+  if (_anchorType == 0) {
+    verticalConstraint = [self.view.bottomAnchor constraintEqualToAnchor:layoutGuide.bottomAnchor
+                                                                constant:_anchorOffset];
   } else {
-    y = _anchorOffset;
+    verticalConstraint = [self.view.topAnchor constraintEqualToAnchor:layoutGuide.topAnchor
+                                                             constant:_anchorOffset];
   }
-  self.adView.frame = (CGRect){{x, self.adView.frame.origin.y}, self.adView.frame.size};
+
+  [NSLayoutConstraint activateConstraints:@[
+    verticalConstraint,
+    [self.view.centerXAnchor constraintEqualToAnchor:layoutGuide.centerXAnchor
+                                            constant:_horizontalCenterOffset],
+    [self.view.leftAnchor constraintGreaterThanOrEqualToAnchor:layoutGuide.leftAnchor],
+    [self.view.rightAnchor constraintLessThanOrEqualToAnchor:layoutGuide.rightAnchor],
+  ]];
+}
+
+- (void)activateConstraintParentView:(UIView *_Nonnull)parentView
+                   verticalAttribute:(NSLayoutAttribute)verticalAttribute {
+  [parentView
```

Comment:
The indentation is slightly different here than in the other ones. Is that intended?

Reply:
This was done automatically by the clang formatter.